Mathlib/Order/Partition/Finpartition.lean

```diff
@@ -586,6 +586,24 @@ theorem mem_part_ofSetoid_iff_rel {s : Setoid α} [DecidableRel s.r] {b : α} :
   simp only [← hc, mem_univ, mem_filter, true_and] at this ⊢
   exact ⟨s.trans (s.symm this), s.trans this⟩
 
+section Two
+
+/-- A bipartition defined by one part, the other part being its complement.
+Both parts must be non-empty. -/
+def bipartition (h1 : s ≠ ∅) (h2 : s ≠ univ) : Finpartition (univ : Finset α) where
+  parts := {s, sᶜ}
```

Comment:
In fact, this definition Just Works already

Reply:
So why have `properBipartition` as a separate definition?

Reply:
I couldn't get `bipartition` to work with `parts := {s, sᶜ}` and no conditions on `s`.